ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManagerImpl.java

```diff
@@ -342,8 +342,12 @@ public void onPackagerStatusFetched(final boolean packagerIsRunning) {
                       public void run() {
                         if (packagerIsRunning) {
                           mDevSupportManager.handleReloadJS();
-                        } else {
+                        } else if (mDevSupportManager.hasUpToDateJSBundleInCache()) {
```

Comment:
I don't think recreateReactContextInBackgroundFromBundleFile falls back to the cached bundle. But maybe it doesn't correctly handle the case when mJSBundleFile is null? Maybe that's what we should be checking for?